proxy-socket-core/src/main/java/net/airvantage/proxysocket/core/v2/ProxyProtocolV2Decoder.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * MIT License
+ * Copyright (c) 2025 Semtech
+ */
+package net.airvantage.proxysocket.core.v2;
+
+import net.airvantage.proxysocket.core.ProxyProtocolParseException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Dependency-free Proxy Protocol v2 utilities (validate/parse/build).
+ */
+public final class ProxyProtocolV2Decoder {
+    private ProxyProtocolV2Decoder() {}
+
+    private static final byte[] SIG = "\r\n\r\n\0\r\nQUIT\n".getBytes(StandardCharsets.ISO_8859_1);
+    private static final int IPV4_ADDR_LEN = 4;
+    private static final int IPV6_ADDR_LEN = 16;
+    private static final int UNIX_ADDR_LEN = 216;
+    private static final int PORT_LEN = 2;
+    private static final int TLV_HEADER_LEN = 3;
+
+    public static ProxyHeader parse(byte[] data, int offset, int length) throws ProxyProtocolParseException {
+        if (data == null || offset < 0 || length < 0) throw new ProxyProtocolParseException("Null data");
```

Comment:
I'm not sure to get why offset < 0  or length < 0 is null data maybe invalid argument ? 

data= null ok or data.lenght == 0 or length == 0  could be null data but this sounds also to be invalid argument... 

Maybe we should use : `IllegalArgumentException` for this kind of error and keep ProxyProtocolParseException when this is really a parsing error ?